### DISCUSSION TITLE
Sunrise Sunset feature for activetimerange

### DISCUSSION THREAD

**Discussion by Yonny24** [VorlonCD/bi-aidetection#289] [Ideas]
Hi @VorlonCD, 

Hope you are well.

I know I'm not alone in wanting this extra feature but would it be at all possible to add a sunrise/sunset time for the ActiveTimeRange option on the deepstack server options? I have it fixed but need to adjust it weekly.

This would be fantastic. No idea how complex it would be to add this though.

Thanks

**Comment by VorlonCD** [comment]
Its been in there for a while, just use "sunset-sunrise" or "sunrise-sunset".   I think dusk-dawn works also and uses a slightly different calc

**Comment by Yonny24** [comment]
Oh brilliant!

So I just change the timeframe to sunset-sunrise?

![image](https://user-images.githubusercontent.com/64428686/140750387-c4187b47-159c-4223-80c5-b4b73199969d.png)


![image](https://user-images.githubusercontent.com/64428686/140750459-43e83db2-95a9-40bc-8a7f-5e9f175af751.png)

**Comment by VorlonCD** [comment]
Yes.   And IF a BI install is detected by AITOOL it will automatically use its Lat/long GPS coords for better results.   BI > Settings > Schedule tab.   Otherwise, you can edit AITOOL.SETTINGS.JSON and change 'LocalLatitude' and 'LocalLongitude'.   Set to what you find here:  https://www.gps-coordinates.net/

I think dusk-dawn uses the 6 degrees mark rather than horizon:
![image](https://user-images.githubusercontent.com/7512113/140753095-e6f4ceb0-c386-46a3-b9bd-a088aa3d52a6.png)

**Comment by Yonny24** [comment]
No options of an offset yet though? Asking for the moon on a stick sorry.

**Reply by VorlonCD** [reply]
@Yonny24 - Offset?

**Comment by johngianni** [comment]
Thank you so much @VorlonCD . Your app is still so much ahead of the standard BI. I hope you will continue to improve it.

**Reply by Yonny24** [reply]
> 
> 
> Thank you so much @VorlonCD . Your app is still so much ahead of the standard BI. I hope you will continue to improve it.

I agree. It's already come along so far in the last 11 months since I've been using it.
Great work!

**Comment by johngianni** [comment]
Just a little contribution. On an Italian Windows with COMMA decimal separator, the lat and long conversion from the registry key is resulting in the wrong number, thus giving an error in the log (number must me < 90 and > -90 . Please double check the ToDouble conversion in the file blueiris.cs at line 325. Suggested change is: 

`
double localCultreResult;
                        double.TryParse(key.GetValue("latitude", "39.809734").ToString(), System.Globalization.NumberStyles.Any, System.Globalization.CultureInfo.InvariantCulture, out localCultreResult);
                        this.Latitude = localCultreResult;`
